api/db/models/petition.py

```diff
@@ -0,0 +1,31 @@
+from sqlalchemy import Column, Integer, String, ForeignKey, DateTime, Enum
+from sqlalchemy.orm import relationship
+from sqlalchemy.sql import func
+from ..base_class import Base
+from .supporter_petitions import supporter_petitions
+import enum
+
+
+class Status(enum.Enum):
+    active = "active"
+    victory = "victory"
+    closed = "closed"
+
+
+class Petition(Base):
+    __tablename__ = "petition"
+
+    id = Column(Integer, primary_key=True, index=True)
+    title = Column(String, nullable=False)
+    description = Column(String, nullable=False)
+    image = Column(String)
+    decision_maker = Column(String, nullable=False)
+    location = Column(String, nullable=False)
+    creation_time = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
+    status = Column(Enum(Status), default=Status.active)
+    supporters = relationship("User",
+                              secondary=supporter_petitions,
+                              back_populates="petitions")
+    owner_id = Column(Integer, ForeignKey("user.id"), nullable=False)
+    owner = relationship("User", back_populates="own_petitions")
+    complains = relationship("Complain", back_populates="petition")
```

Comment:
I think you also need an associate table here to implement many-to-many relations.

Reply:
Here is implemented a one-to-many relation between petitions and complaints tables because a few specifics complaints could rely on one specific petition the same complaint from the user can not rely on some other petition

Reply:
Sorry, you’re right